jkube-kit/common/src/test/java/org/eclipse/jkube/kit/common/util/OpenshiftHelperStatusTest.java

```diff
@@ -0,0 +1,121 @@
+/**
+ * Copyright (c) 2019 Red Hat, Inc.
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at:
+ *
+ *     https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.jkube.kit.common.util;
+
+import org.junit.Test;
+import org.junit.experimental.runners.Enclosed;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+import java.util.Arrays;
+import java.util.Collection;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+@RunWith(Enclosed.class)
+public class OpenshiftHelperStatusTest {
+    @RunWith(Parameterized.class)
+    public static class Finished {
+        @Parameterized.Parameters(name = "{0}: isFinished({1})={2}")
+        public static Collection<Object[]> data() {
+            return Arrays.asList(new Object[][]{
+                    {"IsCompleteTrue", "Complete", true},
+                    {"IsErrorTrue", "Error", true},
+                    {"IsCancelledTrue", "Cancelled", true},
+                    {"IsNotCompleteFalse", "not Complete", false}
+            });
+        }
+
+        @Parameterized.Parameter
+        public String description;
+        @Parameterized.Parameter(1)
+        public String input;
+        @Parameterized.Parameter(2)
+        public Boolean expected;
+
+        @Test
+        public void testIsFinishedUsingParametrizedTest() {
+            assertThat(OpenshiftHelper.isFinished(input)).isEqualTo(expected);
```

Comment:
You might be able to avoid all these duplicate tests by using a `BooleanSupplier` parameter

Reply:
polite ping, are you still working on this?

Reply:
yes rohan, sorry for the delay, Allow me to do it today!